lib/src/cmaf/ham/mapper/hls/mapHlsToHam.ts

```diff
@@ -61,30 +102,29 @@ function mapHlsToHam(manifest: Manifest) {
 		} as SwitchingSet);
 	}
 
-	selectionSets.push({
-		id: uuid(),
-		switchingSets: audioSwitchingSets,
-	} as SelectionSet);
+	return audioSwitchingSets;
+}
 
+function _subtitleGroupsToSwitchingSets(
+	mediaGroupsSubtitles: any,
+	manifestPlaylists: Manifest[],
+): SwitchingSet[] {
 	const subtitleSwitchingSets: SwitchingSet[] = [];
 
-	// Add selection set of type subtitles
 	for (const subtitle in mediaGroupsSubtitles) {
 		const attributes = mediaGroupsSubtitles[subtitle];
 		const textTracks: TextTrack[] = [];
 		const keys = Object.keys(attributes);
 		const { language, uri } = attributes[keys[0]];
-		const subtitleParsed = parseM3u8(
-			manifestPlaylists[currentPlaylist++].manifest,
-		);
+		const subtitleParsed = parseM3u8(manifestPlaylists.shift()!.manifest);
 		const segments: Segment[] = _formatSegments(subtitleParsed?.segments);
 		const targetDuration = subtitleParsed?.targetDuration;
 		textTracks.push({
 			id: subtitle,
-			type: 'TEXT',
+			type: 'text',
 			name: uri,
 			codec: '',
-			duration: targetDuration,
+			duration: targetDuration * segments.length,
```

Comment:
This change was made for playing video in DASH?
Even though it works fine in HLS, we were using this value for EXT-X-TARGET-DURATION.

It is not current complication, maybe we should clarify

Reply:
Maybe a GAP here? DASH is using duration as the total duration of the track. But target duration in HLS is the duration for a single segment, right?

Reply:
Thinking more on it, maybe in the conversion HAM -> HLS we can calculate the Target duration by doing `duration/segments.length`

Reply:
I agree, this seems to resolve this issue. 
Even though duration is the same for both as you mention , both dash and hls use it differently.

Reply:
I updated the TARGETDURATION in the map to HLS